### DISCUSSION TITLE
Licence for SchemaCrawler with/without JDBC drivers

### DISCUSSION THREAD

**Discussion by mark-burch** [schemacrawler/SchemaCrawler#499] [Q&A] [answered]
Hello @sualeh ! First let me say that SchemaCrawer is fantastic and has helped me tremendously already. Just a quick questions about the licensing for it:

1.  The website mentions that the packages that include JDBC drivers are licensed under GPL. Is this true for all plugins? Oracle has quite different licensing requirements as well as Postgres which is more lenient than GPL.
2. As far as I can see, to get support for a database (say Postgres) you need the schemacrawler-postgres plugin and that comes with the JDBC driver. So how does one use the driver-less library if it can't interface with any database without a respective DB plugin JAR? Are the plugins needed only for a subset of operations?
3. If one were to include the relevant plugin (say schemcrawler-oracle) but exclude the Oracle JDBC driver dependency - does that mean I can maintain an LGPL or EPL license?

I am really interested in using SchemaCrawler as an external library. Thank you so much in advance and once again - thanks for this amazing piece of work.

**Comment by sualeh** [comment, marked as answer]
@alwaysbemark, I am so glad you enjoy using SchemaCrawler! Let me explain my intention behind the licensing terms I have chosen.

> 1. The website mentions that the packages that include JDBC drivers are licensed under GPL. Is this true for all plugins? Oracle has quite different licensing requirements as well as Postgres which is more lenient than GPL.

I am not sure that what is on the website the best to express it, but what I am trying to say here is that if you want to redistribute the JDBC drivers to your users, you should do that by downloading the drivers independently of the SchemaCrawler download, or have your customers download them. On the other hand, if you want to use SchemaCrawler as a stand-alone command-line tool for yourself, you can use it with the GPL license. For more clarity, please see the answers to your other questions.


> 2. As far as I can see, to get support for a database (say Postgres) you need the schemacrawler-postgres plugin and that comes with the JDBC driver. So how does one use the driver-less library if it can't interface with any database without a respective DB plugin JAR? Are the plugins needed only for a subset of operations?

You can use SchemaCrawler without the corresponding SchemaCrawler plugin, but the functionality will be limited. Using the SchemaCrawler plugin for a database gives a much richer experience. You will need a JDBC driver in either case, though. If you are using it yourself, then the SchemaCrawler download comes with a lot of JDBC drivers. If you redistribute SchemaCrawler to someone else, then you should do it without the JDBC drivers obtained from the SchemaCrawler download. You can redistribute the SchemaCrawler plugins, though.


> 3. If one were to include the relevant plugin (say schemcrawler-oracle) but exclude the Oracle JDBC driver dependency - does that mean I can maintain an LGPL or EPL license?

Yes. You can redistribute any SchemaCrawler jar (including plugins) under either an LGPL or EPL license, as long as you do not redistribute the JDBC drivers that you downloaded along with SchemaCrawler. You can make modifications to SchemaCrawler (including plugins) under the terms of LGPL or EPL as well. If you need to work within some other custom licensing terms, this can be arranged using a Professional Support Package, as described on the [Consulting page on the website](https://www.schemacrawler.com/consulting.html).

**Reply by mark-burch** [reply]
That settles it for me, thank you so much!
